cloudslang-all/src/main/java/io/cloudslang/lang/api/SlangImpl.java

```diff
@@ -117,6 +117,12 @@ public void subscribeOnAllEvents(ScoreEventListener eventListener) {
         eventTypes.add(ScoreLangConstants.EVENT_INPUT_END);
         eventTypes.add(ScoreLangConstants.EVENT_OUTPUT_START);
         eventTypes.add(ScoreLangConstants.EVENT_OUTPUT_END);
+        eventTypes.add(ScoreLangConstants.EVENT_BRANCH_START);
+        eventTypes.add(ScoreLangConstants.EVENT_BRANCH_END);
+        eventTypes.add(ScoreLangConstants.EVENT_ASYNC_LOOP_INPUT_START);
```

Comment:
I introduced them to distinguish between the split-join steps and normal tasks - event scenario:

- when execute the split step of an async loop we fire an EVENT_ASYNC_LOOP_INPUT_END event which contains the result list of *for* : `for: value in values` e.g. [1,2,3] - this is different content than a normal INPUT EVENT
- then we have EVENT_BRANCH_START events with the specific bound input from this list: [1,2,3]
- events from branch (input, output ...)
- EVENT_BRANCH_END events: with the branch return values
- EVENT_ASYNC_LOOP_OUTPUT_END: aggregate and navigate values (this is similar to a normal output event)

This is also handy in other situations where we want to detect async loop:

- system tests (BranchAggregatorListener, JoinAggregatorListener)
- CLI: print task names